From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Mon, 14 Nov 2022 13:39:39 -0600
Subject: [PATCH] windowscodecs: HACK: Avoid using __builtin_frame_address()
 for setjmp().

This is a workaround to compiler bug (observed with Mingw / Gcc 10.x).

CW-Bug-Id: #21563
Valve-commit: 7dc69f2a4991ccc26dd34b9c50805d03b1a23c38
Picked: clean
Class: valve
Evidence: docs/evidence/2026-09-28-wine-proton-rebase.md
Offered-upstream: no
---
 dlls/windowscodecs/wincodecs_private.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/windowscodecs/wincodecs_private.h b/dlls/windowscodecs/wincodecs_private.h
index 2608b72..d699e10 100644
--- a/dlls/windowscodecs/wincodecs_private.h
+++ b/dlls/windowscodecs/wincodecs_private.h
@@ -467,4 +467,9 @@ extern HRESULT CommonDecoder_CreateInstance(struct decoder *decoder,
 extern HRESULT CommonEncoder_CreateInstance(struct encoder *encoder,
     const struct encoder_info *encoder_info, REFIID iid, void** ppv);
 
+#ifdef _WIN64
+#undef setjmp
+#define setjmp(buf) _setjmpex(buf, NULL)
+#endif
+
 #endif /* WINCODECS_PRIVATE_H */
